### ISSUE TITLE
Malformed model object stuctures breaks stuff, including MPD server

### ISSUE THREAD

**Issue by kingosticks** [mopidy/mopidy#865]
If you add add a track to the tracklist that has an an empty string for the album name, the Track instance within Mopidy has no `artist.name` field. But the MPD translator's `track_to_mpd_format` function will try to access this non-existent field and raise an exception. This breaks the MPD `playlist` command and prevents my MPD client (gmpc) from connecting to Mopidy.

Steps to reproduce:
* Add to the tracklist:
```
curl -X POST -H Content-Type:application/json -d '{
  "method": "core.tracklist.add",
  "jsonrpc": "2.0",
  "params": [[
{"__model__":"Track","name":"Foo","uri":"local:some:uri","album":{"uri":"x","name":""}}]]
  ,
  "id": 1
}' http://localhost:6680/mopidy/rpc
```
* Send MPD command and get a timeout: `mpc playlist`

Traceback reports "AttributeError: 'dict' object has no attribute 'name'".
Full traceback at http://dpaste.com/0DHKT27

**Comment by kingosticks** [comment]
With the 1.0 API, it probably makes most sense to add to tracklist by uri rather than track whenever possible. That should remove the issue in some cases.

**Comment by adamcik** [comment]
Oh yes, I really hope so. As I can't really blame people for getting this wrong with the old APIs. Next step after this model validation fixing and more robust backend calls is likely to add type checks on core calls. And hopefully we can at least mostly trust backends to gives us the right types - though coding defensively there might not be a bad idea either.

**Comment by jodal** [comment]
Fixed by the merge of PR #1105.
